### PR TITLE
Add arpack

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: recipes/arpack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - openblas 0.2.18*
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libarpack.so  # [linux]
+    - test -f ${PREFIX}/lib/libarpack.dylib  # [osx]
+
+about:
+  home: https://github.com/opencollab/arpack-ng
+  license: BSD
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: What version of BSD (e.g. `BSD 3-Clause`)?

File: recipes/arpackpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  run:
+    - arpack
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/arpack++/arpackf.h
+
+about:
+  home: http://www.ime.unicamp.br/~chico/arpack++/
+  license: BSD
```
Comment (jakirkham) [inline recipes/arpackpp/meta.yaml]: What version of BSD (e.g. `BSD 3-Clause`)?

File: recipes/arpackpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+  script:
+    - install -d ${PREFIX}/include/arpack++
+    - install include/*.h ${PREFIX}/include/arpack++
+
+requirements:
+  run:
+    - arpack
```
Comment (jakirkham) [inline recipes/arpackpp/meta.yaml]: Just to confirm, this is header only?

File: recipes/arpack/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  sha256: 69e9fa08bacb2475e636da05a6c222b17c67f1ebeab3793762062248dd9d842f
+
+build:
+  number: 200
+  features:
+    - blas_{{ variant }}
+  skip: true  # [win]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Please add `toolchain`.

File: recipes/arpack/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - gcc
+    - cmake
+    - blas 1.1 {{ variant }}
+    - openblas 0.2.18*
+
+  run:
+    - libgcc
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Please change to `libgfortran`.

File: recipes/arpack/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    - blas 1.1 {{ variant }}
+    - openblas 0.2.18*
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libarpack.so  # [linux]
+    - test -f ${PREFIX}/lib/libarpack.dylib  # [osx]
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Any tests for headers? Static libraries?

File: recipes/arpack/build.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  SO_EXT='.so'
+fi
+
+
+mkdir -p build && cd build
+
+cmake \
+  -DCMAKE_PREFIX_PATH=${PREFIX} \
+  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+  -DBUILD_SHARED_LIBS=ON \
```
Comment (jakirkham) [inline recipes/arpack/build.sh]: Could we please enable static libraries too?

File: recipes/arpack/build.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON \
+  ..
+
+make -j${CPU_COUNT}
+cp lib/libarpack${SO_EXT} ${PREFIX}/lib
+DYLD_FALLBACK_LIBRARY_PATH=${PREFIX}/lib make check -j${CPU_COUNT}
```
Comment (jakirkham) [inline recipes/arpack/build.sh]: No `make install`?

File: recipes/arpackpp/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+{% set version = "1.2" %}
+
+package:
+  name: arpackpp
+  version: {{ version }}
+
+source:
+  fn: arpackpp-{{ version }}.tar.gz
+  url: http://www.ime.unicamp.br/~chico/arpack++/arpack++.tar.gz
```
Comment (jakirkham) [inline recipes/arpackpp/meta.yaml]: Do they have versioned releases? This seems a bit fragile.

File: recipes/arpack/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
+{% set version = "3.4.0" %}
+{% set variant = "openblas" %}
+
+package:
+  name: arpack
+  version: {{ version }}
+
+source:
+  fn: arpack-{{ version }}.tar.gz
+  url: https://github.com/opencollab/arpack-ng/archive/{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Are we sure this is the official version? If so, what convinces us?

File: recipes/arpack/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  features:
+    - blas_{{ variant }}
+  skip: true  # [win]
+
+requirements:
+  build:
+    - gcc
+    - cmake
+    - blas 1.1 {{ variant }}
+    - openblas 0.2.18*
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Please change to `0.2.18|0.2.18.*`.

File: recipes/arpack/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+  run:
+    - libgcc
+    - blas 1.1 {{ variant }}
+    - openblas 0.2.18*
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Please change to `0.2.18|0.2.18.*`.

File: recipes/arpack/meta.yaml (hunk 10 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: It's the one packaged in debian.

File: recipes/arpack/meta.yaml (hunk 6 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: there's no headers

File: recipes/arpackpp/meta.yaml (hunk 9 above)
Comment (jschueller) [inline recipes/arpackpp/meta.yaml]: nope

File: recipes/arpackpp/meta.yaml (hunk 3 above)
Comment (jschueller) [inline recipes/arpackpp/meta.yaml]: yes

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jschueller) [inline recipes/arpack/build.sh]: I prefer not too

File: recipes/arpack/build.sh (hunk 8 above)
Comment (jschueller) [inline recipes/arpack/build.sh]: was add, in master only

**Comment by jschueller** [comment]
@jakirkham I dont know if conda supports non-text chars in the name : could the name be arpack++ instead of arpackpp ?

File: recipes/arpack/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Really? That seems strange. How does one build against ARPACK then?

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jakirkham) [inline recipes/arpack/build.sh]: Why not? Some people prefer to use static libraries.

File: recipes/arpack/meta.yaml (hunk 6 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: It's like lapack I guess. No need for headers.

File: recipes/arpack/meta.yaml (hunk 4 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: ok

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jschueller) [inline recipes/arpack/build.sh]: It's either shared or static using cmake.

File: recipes/arpack/meta.yaml (hunk 11 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: ok

File: recipes/arpack/meta.yaml (hunk 1 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: added

File: recipes/arpack/meta.yaml (hunk 5 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: ok

File: recipes/arpack/meta.yaml (hunk 12 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: ok

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jakirkham) [inline recipes/arpack/build.sh]: Maybe we could just have `for`-loop around the build and try both.

Comment (jschueller) [inline recipes/arpack/build.sh]: That'd be doable, but frankly I don't see the point of providing statics.
Do we want all library packages to provides both then ?

**Comment by jschueller** [comment]
is it good to go ?

**Comment by jakirkham** [comment]
It seems there are a bunch of comments that are still not addressed. Could you please address them? Then will take another look.

File: recipes/arpackpp/meta.yaml (hunk 2 above)
Comment (jschueller) [inline recipes/arpackpp/meta.yaml]: 3-clause, ok

**Comment by jschueller** [comment]
@jakirkham they're all adressed, except for the static libs remark which I do not agree

File: recipes/arpack/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: This doesn't appear to be addressed.

File: recipes/arpack/meta.yaml (hunk 11 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: This doesn't appear to be addressed.

File: recipes/arpack/meta.yaml (hunk 5 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: This doesn't appear to be addressed.

File: recipes/arpack/meta.yaml (hunk 12 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: This doesn't appear to be addressed.

File: recipes/arpack/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: Done below.

File: recipes/arpackpp/meta.yaml (hunk 9 above)
Comment (jakirkham) [inline recipes/arpackpp/meta.yaml]: I'm also seeing this [webpage]( http://reuter.mit.edu/software/arpackpatch/ ) for it, which points at this [repo]( https://github.com/m-reuter/arpackpp ).

File: recipes/arpackpp/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  run:
+    - arpack
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/arpack++/arpackf.h
+
+about:
+  home: http://www.ime.unicamp.br/~chico/arpack++/
+  license: BSD 3-Clause
```
Comment (jakirkham) [inline recipes/arpackpp/meta.yaml]: Please add the license file.

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jakirkham) [inline recipes/arpack/build.sh]: That has been the status quo is trying to provide both dynamic and static libraries so people can chose what is appropriate for their use case.

File: recipes/arpack/meta.yaml (hunk 10 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: AFAICT you are correct to use this. Arch Linux and Gentoo both use this too. Not sure about other Linux distros though.

File: recipes/arpackpp/meta.yaml (hunk 9 above)
Comment (jschueller) [inline recipes/arpackpp/meta.yaml]: yeah, it should be this one

File: recipes/arpack/meta.yaml (hunk 4 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: Oh sorry I thought it was.

File: recipes/arpack/meta.yaml (hunk 11 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: that too.

File: recipes/arpackpp/meta.yaml (hunk 13 above)
Comment (jschueller) [inline recipes/arpackpp/meta.yaml]: done

File: recipes/arpack/meta.yaml (hunk 12 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: ok

File: recipes/arpack/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+about:
+  home: https://github.com/opencollab/arpack-ng
+  license: BSD 3-Clause
+  license_file: COPYING
+  summary: Fortran77 subroutines designed to solve large scale eigenvalue problems
+
+extra:
+  recipe-maintainers:
+    - jschueller
+
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: nit: Please make one terminal newline just like `arpackpp/meta.yaml`.

File: recipes/arpack/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: No worries. Thanks for addressing.

File: recipes/arpack/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  run:
+    - libgfortran
+    - blas 1.1 {{ variant }}
+    - openblas 0.2.18|0.2.18.*
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libarpack.a  # [unix]
+    - test -f ${PREFIX}/lib/libarpack.so  # [linux]
+    - test -f ${PREFIX}/lib/libarpack.dylib  # [osx]
```
Comment (jakirkham) [inline recipes/arpack/meta.yaml]: 👍

File: recipes/arpack/meta.yaml (hunk 14 above)
Comment (jschueller) [inline recipes/arpack/meta.yaml]: done

File: recipes/arpack/build.sh (hunk 7 above)
Comment (jschueller) [inline recipes/arpack/build.sh]: ok

**Comment by jakirkham** [comment]
LGTM. Thanks @jschueller. Will merge once it passes.

**Comment by jakirkham** [comment]
Sorry wrong button. 😳

**Comment by jschueller** [comment]
did you cancel the windows build ?

**Pull request merged**